### PR TITLE
fix(console): Add `idp` name validation on keystroke

### PR DESCRIPTION
**Pull request by yasinmiran**
### Related Issues
(None)

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [x] UX/UI review done on final implementation
- [ ] Documentation provided. (Add links)
- [ ] Unit tests provided. (Add links if any)
- [ ] Integration tests provided. (Add links if any)

### Related PRs
(None)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [x] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx [hunk 1]
```diff
@@ -118,10 +119,24 @@ export const EnterpriseIDPCreateWizard: FC<EnterpriseIDPCreateWizardProps> = (
...
 
     const dispatch = useDispatch();
     const { t } = useTranslation();
 
+    useEffect(() => {
+        setIsIDPListLoading(true);
+        getIdentityProviderList(null, null, null)
+            .then((response) => {
+                setIdPList(response.identityProviders);
+            }).catch((error) => {
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx:133]:
```suggestion
            })
            .catch((error) => {
```

File: apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx [hunk 2]
```diff
@@ -118,10 +119,24 @@ export const EnterpriseIDPCreateWizard: FC<EnterpriseIDPCreateWizardProps> = (
...
+            handleGetIDPListCallError(error);
+        }).finally(() => {
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx:135]:
```suggestion
        })
        .finally(() => {
```

File: apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx [hunk 3]
```diff
@@ -326,6 +350,9 @@ export const EnterpriseIDPCreateWizard: FC<EnterpriseIDPCreateWizardProps> = (
         <WizardPage validate={ (values: any) => {
             const errors: FormErrors = {};
             errors.name = composeValidators(required, length(IDP_NAME_LENGTH))(values.name);
+            if (isIdpNameAlreadyTaken(values.name)) {
+                errors.name = "An identity provider already exists with this name.";
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx:354]: Shall we add the `i18n` placeholder.

File: apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx [hunk 4]
```diff
@@ -814,15 +847,18 @@ export const EnterpriseIDPCreateWizard: FC<EnterpriseIDPCreateWizardProps> = (
...
-                        <Wizard2
-                            ref={ wizardRef }
-                            initialValues={ initialValues }
-                            onSubmit={ handleFormSubmit }
-                            uncontrolledForm={ true }
-                            pageChanged={ (index: number) => setCurrentWizardStep(index) }
-                            data-testid={ testId }>
-                            { resolveWizardPages() }
-                        </Wizard2>
+                        { !isIDPListLoading ?
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx:850]: Lets use the formatting in [DEVELOPER](https://github.com/wso2/identity-apps/blob/master/docs/DEVELOPER.md#formatting) guide for ternary operators.

Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx]: Addressed this in commit 8e64d82

File: apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx (hunk 3 above)
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx]: This is addressed in commit d9d648b

File: apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx (hunk 1 above)
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx]: Addressed this in commit 55d2628.

**Review by brionmario — APPROVED** [review]

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx [hunk 5]
```diff
@@ -198,6 +215,15 @@ export const EnterpriseIDPCreateWizard: FC<EnterpriseIDPCreateWizardProps> = (
         ]
     };
 
+    const isIdpNameAlreadyTaken = (userInput: string): boolean => {
+        if (idpList?.length > 0) {
+            return idpList
+                .reduce((set, { name }) => set.add(name), new Set<string>())
+                .has(userInput);
+        }
+        return false;
```
Comment (thivi) [inline apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx:224]: Shall we add a line before the return statement?

File: apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx [hunk 6]
```diff
@@ -814,15 +852,18 @@ export const EnterpriseIDPCreateWizard: FC<EnterpriseIDPCreateWizardProps> = (
...
-                            initialValues={ initialValues }
-                            onSubmit={ handleFormSubmit }
-                            uncontrolledForm={ true }
-                            pageChanged={ (index: number) => setCurrentWizardStep(index) }
-                            data-testid={ testId }>
-                            { resolveWizardPages() }
-                        </Wizard2>
+                        { !isIDPListLoading
+                            ? <Wizard2
+                                ref={ wizardRef }
```
Comment (thivi) [inline apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx:857]: I think `WizardTwo` will be a better name than `Wizard2`. WDYT?

**Pull request merged**